tests/robot-tests/tests/public_api/public_api_dataset_cancel_and_removal.robot

```diff
@@ -0,0 +1,160 @@
+*** Settings ***
+Library             ../libs/admin_api.py
+Resource            ../libs/admin-common.robot
+Resource            ../libs/admin/manage-content-common.robot
+Resource            ../libs/public-common.robot
+Resource            ../libs/public-api-common.robot
+
+Force Tags          Admin    Local    Dev    AltersData
+
+Suite Setup         user signs in as bau1
+Suite Teardown      user closes the browser
+
+
+
+*** Variables ***
+${PUBLICATION_NAME}=    UI tests - public api restricted %{RUN_IDENTIFIER}
+${RELEASE_NAME}=        Financial year 3000-01
+${SUBJECT_NAME_1}=      UI test subject 1
+${SUBJECT_NAME_2}=      UI test subject 2
+
+
+
+*** Test Cases ***
+Create publication and release
+   ${PUBLICATION_ID}=    user creates test publication via api    ${PUBLICATION_NAME}
+    user creates test release via api    ${PUBLICATION_ID}    FY    3000
+    user navigates to draft release page from dashboard    ${PUBLICATION_NAME}
+    ...    ${RELEASE_NAME}
+
+Verify release summary
+    user checks page contains element    xpath://li/a[text()="Summary" and contains(@aria-current, 'page')]
+    user verifies release summary    Financial year    3000-01    Accredited official statistics
+
+Upload datafiles
+    user uploads subject    ${SUBJECT_NAME_1}    seven_filters.csv    seven_filters.meta.csv
+    user uploads subject    ${SUBJECT_NAME_2}    tiny-two-filters.csv    tiny-two-filters.meta.csv
+
+Add data guidance to subjects
+    user clicks link    Data and files
+    user waits until h2 is visible    Add data file to release
+
+    user clicks link    Data guidance
+    user waits until h2 is visible    Public data guidance
+
+    user waits until page contains element    id:dataGuidance-dataFiles
+    user waits until page contains accordion section    ${SUBJECT_NAME_1}
+
+    user enters text into data guidance data file content editor    ${SUBJECT_NAME_1}
+    ...    ${SUBJECT_NAME_1} Main guidance content
+
+    user enters text into data guidance data file content editor    ${SUBJECT_NAME_2}
+    ...    ${SUBJECT_NAME_2} Main guidance content
+
+Save data guidance
+    user clicks button    Save guidance
+
+User clicks on 'Cancel' button while after selecting 1st API dataset
+    user scrolls to the top of the page
+    user clicks link    API data sets
+    user waits until h2 is visible    API data sets
+
+    user clicks button    Create API data set
+    ${modal}=    user waits until modal is visible    Create a new API data set
+    user chooses select option    id:apiDataSetCreateForm-releaseFileId   ${SUBJECT_NAME_1}
+    user clicks button by index    Cancel    3
```

Comment:
Just noting that I think selecting like this can make the test fragile, as if a new cancel button gets added to the page, this test will break. 

But I don't think this is a big deal / worth the time to change, so no need to change this. I'm more just highlighting that we want to make the tests as robust against future changes where feasible. Generally in this situation I'd be inclined to add a way to select a parent element first and then select the cancel button within that parent element. 

I can imagine there are reasons for doing it the way you have - like maybe selecting parent elements first is a source of flaky tests.

Reply:
Done